### DISCUSSION TITLE
Automation not beeing triggered

### DISCUSSION THREAD

**Discussion by JCV67** [Budibase/budibase#13522] [Help] [answered]
Self hosted v2.33.9 | Postgres external tables
Hello all,

Created an automation with the trigger "Row Created" Each new row is created by an SQL query, new rows really exist but the automation isn't fired. I can't find out why. In automation history nothing happens.
Appreciate your help
Yours 
José

![imagem](https://github.com/Budibase/budibase/assets/104722906/d7f2f993-5c72-4da2-b00c-7f9f9e26055a)

**Comment by ConorWebb96** [comment]
Hey @JCV67 

Automations will only trigger if done via a production form or using the public api with the production url attached to your tables. If you create rows via sql or use the builder to manipulate the data these automations won't trigger.

Now there are ways to work around this, for example using the `action query result` from your execute query on a button and passing this to a trigger automation. That way when a button is clicked in your production app you basically are getting the same data that you would if you were to use the create row automation trigger.

Hopefully this helps point you in the right direction.

**Reply by JCV67** [reply]
Thank you @ConorWebb96, going to try your suggestion.

**Comment by JCV67** [comment]
It is working now @ConorWebb96, thank you. A suggestion, a switch to enable or disable automation would be useful.

**Reply by ConorWebb96** [reply, marked as answer]
I've gone ahead and created this [feature request](https://github.com/Budibase/budibase/issues/13532) for you.

**Reply by JCV67** [reply]
Great, thank you.
